mapie/utils.py

```diff
@@ -255,7 +257,13 @@ def check_n_features_in(
     >>> print(check_n_features_in(X))
     5
     """
-    n_features_in: int = X.shape[1]
+    if hasattr(X, "shape"):
+        if len(X.shape) <= 1:
+            n_features_in = 1
+        else:
+            n_features_in = X.shape[1]
+    else:
+        n_features_in = _num_features(X)
```

Comment:
Why can't we use `_num_features` in all cases ?

Reply:
Because, for an unknown reason, `_num_features` raise an exception when X is 1-dimensional.